src/Queue.js

```diff
@@ -3,87 +3,116 @@ const { SchedulerInterrupt } = require('./constants/index');
 // A class representation of a process queue that may hold either
 // blocking or non-blocking processes
 class Queue {
-    constructor(scheduler, quantum, priorityLevel, queueType) {
-        this.processes = [];
-        // The queue's priority level; the lower the number, the higher the priority
-        this.priorityLevel = priorityLevel;
-        // The queue's parent scheduler
-        this.scheduler = scheduler;
-        // The queue's allotted time slice; each process in this queue is executed for this amount of time
-        this.quantum = quantum;
-        // A counter to keep track of how much time the queue has been executing so far
-        this.quantumClock = 0;
-        this.queueType = queueType;
-    }
+	constructor(scheduler, quantum, priorityLevel, queueType) {
+		this.processes = [];
+		// The queue's priority level; the lower the number, the higher the priority
+		this.priorityLevel = priorityLevel;
+		// The queue's parent scheduler
+		this.scheduler = scheduler;
+		// The queue's allotted time slice; each process in this queue is executed for this amount of time
+		this.quantum = quantum;
+		// A counter to keep track of how much time the queue has been executing so far
+		this.quantumClock = 0;
+		this.queueType = queueType;
+	}
 
-    // Adds the input process to the queue's list of processes
-    // Also sets the input process's parent queue to this queue
-    // Return the newly added process
-    enqueue(process) {
-        
-    }
+	// Adds the input process to the queue's list of processes
+	// Also sets the input process's parent queue to this queue
+	// Return the newly added process
+	enqueue(process) {
+		this.processes.push(process);
+		process.setParentQueue(this);
+		return process;
```

Comment:
For this one you can do it in 2 lines.  This is strictly stylistic and not necessary just an example/suggestion
`process.setParentQueue(this);`
`return this.process.push(process);`
Your code is 100% fine just a suggestion